### ISSUE TITLE
Status updates not going to ST any

### ISSUE THREAD

**Issue by jjslegacy** [nutechsoftware/alarmdecoder-smartthings#32]
Just realized after the latest updates I Am not getting any updates to my ST setup. I can manually refresh but don’t get status changes or zone updates.  Looked at all the usual but not exactly sure where else to look.

**Comment by jjslegacy** [comment]
I also noticed the webapp sometimes loads the status properly but after a while I see it say please wait loading... and never switches.

**Comment by jjslegacy** [comment]
further inspection seems the web-app is getting all hungup when the upnp notification is enabled.  If I disable it all is good for days.

**Comment by f34rdotcom** [comment]
Q. if disabled your ST still works? I have seen the upnp service cause issues it is easiest to see specifics inside of the info log under /opt/alarmdecoder-web/instances/logs/ at the same time as the problem happens.

**Comment by Nu11u5** [comment]
I have the same issue.  The webapp will stop updating ST (usually with a zone stuck open) and when checking the webapp keypad and live event pages it won’t show event updates.  Restarting the webapp will fix it but eventually it fails again, sometimes within minutes.  I’ll try to do some testing and monitor the log file.

**Comment by markgreinke** [comment]
I'm having this issue too. Rebuilt my AD Net Appliance with latest image too. My Smartthings integrations are now dead in the water. Ugh!

**Comment by f34rdotcom** [comment]
Sorry was offline today with network down. Lots of fun :(
I am usually available at 503 443 2000 x 703 10AM to 10PM PST. Leave a message I may not answer if I dont recognize the phone # and am otherwise preoccupied.

I can go over your setup and see if I broke anything or if you just had a upgrade glitch.

**Comment by markgreinke** [comment]
On a lark I power cycled the Smartthings Hub and it's been updating for the last 2 hours so maybe that was the issue. I will post again tomorrow with a status

**Comment by jjslegacy** [comment]
I have the same issue randomly and fully powering off the SmartThings hub for a few minutes fixes it.  Not sure why though

**Comment by f34rdotcom** [comment]
I can also confirm this happened to me a few times during development. I think all the code changes while running causes the hub to stop listening on its local port. 

Reboot also fixed it. The issue has not returned and seems to only happen when I publish code changes to my hub.

**Comment by markgreinke** [comment]
OK. After restarting my Smarthings hub all is well again. I really thought this issue was the same as what was reported by jjslegacy because I would get updates for a few minutes after restarting the webapp but then they would completely stop. I'm back in business!

**Comment by f34rdotcom** [comment]
Excellent! Keep me posted if you see it go poof again. This problem has a few causes both need work. One idea is to expand on the UPNP plugin page to show connections and last talk times. Still need to work on the UPNP notification and make it asynchronous so it will not stall out other parts of the system when the HUB locks up.

**Comment by jjslegacy** [comment]
I am having this issue again and this time the reboot of the hub isn’t helping.  Seems totally broken now.

**Comment by jjslegacy** [comment]
some log info:

0000]," DISARMED CHIME   Ready to Arm  "]]></last_message_received></panelstate></e:property>
</e:propertyset> [in /opt/alarmdecoder-webapp/ad2web/notifications/types.py:462]
2019-07-18 03:22:59,601 INFO: Event UPNPPush Notification Failed: '' line: 464 [in /opt/alarmdecoder-webapp/ad2web/notifications/types.py:467]
2019-07-18 03:22:59,611 INFO: Event: 120600000000606c02456cf5ec0101010000 [in /opt/alarmdecoder-webapp/ad2web/notifications/types.py:422]
2019-07-18 03:22:59,639 ERROR: Error sending notification for UPNPPush: line: 138 UPNPPushNotification Failed: '' line: 464 [in /opt/alarmdecoder-webapp/ad2web/decoder.py:459]
2019-07-18 03:22:59,668 INFO: _notify_subscribers [in /opt/alarmdecoder-webapp/ad2web/notifications/types.py:447]
2019-07-18 03:22:59,671 INFO: _notify_subscribers: {'4c75c894-a90b-11e9-829a-7f2f4d7d14b9': {'callback': u'<http://10.10.10.106:39500/notify>', 'host': u'10.10.10.3:5000', 'timeout': 1563448960.407198}}
<e:propertyset xmlns:e="urn:schemas-upnp-org:service:AlarmDecoder:1">

**Comment by f34rdotcom** [comment]
Any logs inside smartthings graph page would be helpful as well as logs
from app on AD2 webapp.

Try rebooting the pi and then when it is back up force a refresh on the
bottom of the AD2 st app screen.

**Comment by jjslegacy** [comment]
I see nothing at all in the smartthings graph logs page.  I have rebooted the pi and the smartthings hub.  

The manual refresh from ST App works just fine.

All I really see is this in the logs:

2019-07-18 13:18:22,368 INFO: Event UPNPPush Notification Failed: '' line: 464 [in /opt/alarmdecoder-webapp/ad2web/notifications/types.py:467]
2019-07-18 13:18:22,379 INFO: Event: 1606000000006163024543f531fb456cf5ec03010101 [in /opt/alarmdecoder-webapp/ad2web/notifications/types.py:422]
2019-07-18 13:18:22,409 ERROR: Error sending notification for UPNPPush: line: 138 UPNPPushNotification Failed: '' line: 464 [in /opt/alarmdecoder-webapp/ad2web/decoder.py:459]
2019-07-18 13:19:45,856 INFO: SUBSCRIBE host:10.10.10.3:5000 callback: <http://10.10.10.106:39500/notify> timeout: Second-28800 [in /opt/alarmdecoder-webapp/ad2web/api/views.py:243]
2019-07-18 13:19:45,858 INFO: add_subscriber: ba4d0dde-a91d-11e9-a8d3-b827eba59841 [in /opt/alarmdecoder-webapp/ad2web/notifications/types.py:203]

**Comment by jasonsf** [comment]
I'm having this same issue. WebAPP Log:

`<e:property><panelstate><panel_perimeter_only>False</panel_perimeter_only><panel_chime>False</panel_chime><panel_alarming>False</panel_alarming><panel_battery_trouble>False</panel_battery_trouble><panel_entry_delay_off>False</panel_entry_delay_off><panel_ready>True</panel_ready><panel_fire_detected>False</panel_fire_detected><panel_powered>True</panel_powered><panel_exit>False</panel_exit><panel_panicked>False</panel_panicked><panel_armed_stay>False</panel_armed_stay><panel_type>ADEMCO</panel_type><panel_bypassed>False</panel_bypassed><panel_armed>False</panel_armed><panel_relay_status /><panel_zones_faulted /><last_message_received><![CDATA[[10000001000000003A--],008,[f70200051008001c08020000000000],"****DISARMED****  Ready to Arm  "]]></last_message_received></panelstate></e:property>
</e:propertyset> [in /opt/alarmdecoder-webapp/ad2web/notifications/types.py:462]
2019-07-16 08:55:33,978 INFO: Event UPNPPush Notification Failed: [Errno 113] No route to host line: 464 [in /opt/alarmdecoder-webapp/ad2web/notifications/types.py:467]
2019-07-16 08:55:33,989 INFO: Event: 0451449:0:0:0:0:0:0 [in /opt/alarmdecoder-webapp/ad2web/notifications/types.py:422]
2019-07-16 08:55:34,081 ERROR: Error sending notification for UPNPPush: line: 138 UPNPPushNotification Failed: [Errno 113] No route to host line: 464 [in /opt/alarmdecoder-webapp/ad2web/decoder.py:459]`

**Comment by jjslegacy** [comment]
that looks slightly different as I don't have the " No route to host"

**Comment by jjslegacy** [comment]
Sean so kindly stared at mine long enough until it fixed itself.  It appears although no hard evidence as the issue was on the smartthings side - something stormy in the cloud I guess.

**Comment by jasonsf** [comment]
Magically working on mine too. Pulled the latest web app and library updates also.

**Comment by f34rdotcom** [comment]
Excellent. The latest updated I just did will prevent double ARM events in
ST but if you are using the Raspbian 9 image it wont actually updated the
the API :(

It was a bug in the "oil change" Good news is I just finished testing and
release the latest Raspbian 10 image. I spent a bit of time testing all of
the notifications. Still early but a few have tested it as well as my own
testing and it seems solid. I plan on updating my system at home tomorrow.

Best
 Sean M
